src/distributions/estimate_gmm.h

```diff
@@ -0,0 +1,363 @@
+#ifndef ESTIMATE_GMM_H
+#define ESTIMATE_GMM_H
+
+
+#include <iostream>
+#include <cmath>
+#include <vector>
+#include <sstream>
+
+using namespace std;
+//using namespace edda;
+
+namespace edda
+{
+
+//edda includes
+#include <core/interpolator.h>
+#include <distributions/gaussian_mixture.h>
+
+///////////////////////////////////////////////////////////////////////////////////////////
+///// Traditional EM algorithm section
+
+///////////////////////////////////////////
+/// EM helper function
+int check_convergence(double fmax, double fmin, double ftol)
```

Comment:
Should create a .cpp file and put this function (and all functions below) into the .cpp file.  The header only declares functions, or there will be some linking problem.